opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -744,8 +744,17 @@ namespace Opm
 
         if constexpr (has_foam) {
             // TODO: the application of well efficiency factor has not been tested with an example yet
-            const unsigned gasCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx);
-            EvalWell cq_s_foam = cq_s[gasCompIdx] * this->well_efficiency_factor_;
+            unsigned foamTransportIdx = 0;
+            if (FoamModule::transportPhase() == Phase::GAS)
```

Comment:
Please use `switch` here, 

and also safety check for `has_solvent`. 

Otherwise, when `FoamModule::transportPhase() == Phase::SOLVENT` and `has_solvent == false`, the error message might not point to the real problem.